http-clients/apache-client/src/test/java/software/amazon/awssdk/http/apache/ApacheHttpClientTest.java

```diff
@@ -123,4 +126,23 @@ public void credentialProviderCanBeUsedWithProxy() {
                         .credentialsProvider(Mockito.mock(CredentialsProvider.class))
                         .build();
     }
+
+    @Test
+    public void dnsResolverCanBeUsed() {
```

Comment:
can we add a test that if the caller explicitly sets to null we use the default resolver?

Reply:
yes